### ISSUE TITLE
💅 <TITLE>

### ISSUE THREAD

**Issue by drwpow** [biomejs/biome#2293]
### Environment information

```bash
CLI:
  Version:                      1.6.4
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v21.7.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.5.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 true

Linter:
  Recommended:                  true
  All:                          false
  Rules:                        style/noNonNullAssertion = "off"

Workspace:
  Open Documents:               0
```

### Rule name

lint/style/useImportType

### Playground link

https://biomejs.dev/playground/?code=aQBtAHAAbwByAHQAIAB7ACAAdAB5AHAAZQAgAE0AeQBUAHkAcABlACAAfQAgAGYAcgBvAG0AIAAiAC4ALwBmAG8AbwAiADsACgAKAGMAbwBuAHMAdAAgAHQAOgAgAE0AeQBUAHkAcABlACAAPQAgACIAdAByAHUAZQAiADsA

### Expected result

```ts
import type { MyType } from "foo";
```

should be the same as:

```ts
import { type MyType } from "foo";
```

however the latter isn’t recognized by Biome. I prefer the latter, because:

1. It’s less diff noise if you do have to add or remove a runtime import, and
2. often, VS Code’s <kbd>Tab</kbd> import will make a net-new import statement if there’s an `import type` (not always; I haven’t figured out what causes it, but it overall seems to do a better job with `import { type … }` 🤷 

Do you agree? If those should be parsed the same, happy to assist with a PR 🙂 

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
They aren't the same, unfortunately.

With a specific set of TypeScript options, the second will emit JavaScript that will fail at **runtime**: https://www.typescriptlang.org/play?isolatedModules=true&verbatimModuleSyntax=true&allowSyntheticDefaultImports=true#code/JYWwDg9gTgLgBAbzjAnmApnAsigKmzAXzgDMoIQ4AiEiCKgbgCgg

Since Biome can't understand the tsconfig, we opted for the safest route.

**Comment by drwpow** [comment]
> Also, I'm not what you mean by parsed... Both syntax are parsed and analyzed correctly

Oh my mistake! I was making assumptions, since I got the same lint error whether `{ type MyType }` or `{ MyType }` so I thought it just was ignoring `type` but perhaps not. If it’s an intentional decision that makes more sense.

> Since Biome can't understand the tsconfig, we opted for the safest route.

Ah I didn’t realize! That makes sense. What are your thoughts on just letting the user configure that? For ESLint’s [consistent-type-imports](https://typescript-eslint.io/rules/consistent-type-imports/#fixstyle) rule, for example, the user can specify whether they prefer the default `import type`, vs inline `import { type … }`. That way the decision is on the user, not parsing the tsconfig correctly. Would a configurable option make sense here?

**Comment by Conaclos** [comment]
This is intended. Citing the [rule description](https://biomejs.dev/linter/rules/use-import-type/):

> The rule ensures that all imports used only as a type use a type-only import. It also groups inline type imports into a grouped import type.

The idea behind this behavior is to avoid emitting empty imports such as `import {} from "mod"`. This can lead to runtime error (trying to import a module that doesn't exist) when using a bundler.

> Ah I didn’t realize! That makes sense. What are your thoughts on just letting the user configure that? For ESLint’s [consistent-type-imports](https://typescript-eslint.io/rules/consistent-type-imports/#fixstyle) rule, for example, the user can specify whether they prefer the default import type, vs inline import { type … }. That way the decision is on the user, not parsing the tsconfig correctly. Would a configurable option make sense here?

Why do you need this?

**Comment by drwpow** [comment]
Thanks for the explanation! Actually you’re right; this is working as-intended! Closing because this is working correctly. Thank you!

### Extra explanation

I think the error messages was what was tripping me up, trying to migrate an existing app with all sorts of inconsistent imports (that ESLint’s `consistent-type-imports` ironically wasn’t catching, or I hadn’t configured it correctly), and kudos to Biome for correctly flagging issues I didn’t even know existed! But 2 the error messages from `lint/style/useImportType`, and their intended fix, were tripping me up:

- `All these imports are only used as types` → **You should use `import type …`**
- `Some named imports are only used as types` → **Some named imports are missing the `type` prefix**

Scanning all the error output quickly, I was accidentally reading those as the same error message, and I incorrectly thought Biome was trying to get me to double-up an `import type` along with a runtime import, when really Biome was just going to either use `import type` _or_ `type` named prefix correctly, depending on the situation (I also could have just autofixed it and never seen these, but I was curious what code Biome was suggesting to change, and why).

I’m not suggesting the short descriptions should change; this may just have been my own confusion. Thanks again to the whole Biome team for an amazing tool! I’m so excited to move off ESLint 🙂.

**Comment by Conaclos** [comment]
> that ESLint’s consistent-type-imports ironically wasn’t catching, or I hadn’t configured it correctly

`consistent-type-imports` miss many things. This is why I didn't follow naively their implementation.

> Scanning all the error output quickly, I was initially confusing those 2 problems

Thanks for your detailed explanation. This is the kind of comments that allows improving Biome!
Feel free to open a PR to improve it. Otherwise, I will take a look when I find some time.

**Comment by drwpow** [comment]
> Feel free to open a PR to improve it.

Thanks! I’m not sure if it would be helpful to change `Some named imports are only used as types.` to something like ```Some named imports are missing the `type` prefix``` just yet; will have a think on it (there are other error messages that are similarly-formatted like `The default import and some named imports are only used as types.` which are all cohesive as-written)

But maybe as a start I could update the `use-import-type.md` docs with some more “valid” and “invalid” examples. Will open a PR but feel 100% free to suggest changes and/or revisions!
